Tasks/AzureMysqlDeployment/operations/FirewallOperations.ts

```diff
@@ -0,0 +1,124 @@
+import task = require('vsts-task-lib/task');
+import { ApplicationTokenCredentials} from 'azure-arm-rest/azure-arm-common';
+import { AzureMysqlManagementClient } from 'azure-arm-rest/azure-arm-mysql';
+import { FirewallRule, FirewallAddressRange } from '../models/Firewall';
+import { AzureMysqlTaskParameter } from '../models/AzureMysqlTaskParameter';
+import { FirewallConfiguration } from '../models/FirewallConfiguration';
+import { ISqlClient } from '../sql/ISqlClient';
+import Q = require('q');
+
+export class FirewallOperations {
+
+    private _azureMysqManagementClient: AzureMysqlManagementClient;
+    private _firewallName: string;
+
+    constructor(azureCredentials: ApplicationTokenCredentials, subscriptionId: string) {
+        this._azureMysqManagementClient = new AzureMysqlManagementClient(azureCredentials, subscriptionId);
+    }
+
+    /**
+     * Add firewall rule for particular mysql server
+     * @param serverName          mysql server name
+     * @param firewallRule        firewallRule i.e name and Ip Adress range
+     * @param resourceGroupName   mysql server resource group name
+     * 
+     * @returns operation is success or failure
+     */
+    public async addFirewallRule(serverName: string, firewallRule: FirewallRule, resourceGroupName: string): Promise<void> {
+        let defer = Q.defer<void>();
+        this._azureMysqManagementClient.firewallRules.createOrUpdate(resourceGroupName, serverName, firewallRule.getName(), firewallRule, (error, result, request, response) => {
+            if(error){
+                task.debug("Getting error during adding firewall rule: "+ error);
+                defer.reject(new Error(task.loc("NotAbleToAddFirewallRule", error)));
```

Comment:
defer.reject(task.loc("NotAbleToAddFirewallRule", error));
No need to provide explicit error

Reply:
Sometimes we will get network issue in error to identify for which call we have added NotAbleToAddFirewallRule in error.

Reply:
we could remove new Error statement. 
defer.reject can take a string too